pandas/tests/reshape/concat/test_concat.py

```diff
@@ -395,6 +396,44 @@ def test_concat_keys_with_none(self):
         expected = concat([df0, df0[:2], df0[:1], df0], keys=["b", "c", "d", "e"])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("klass", [range, RangeIndex])
+    def test_concat_preserves_rangeindex(self, klass):
```

Comment:
nit: I _think_ you could parametrize with `include_none=[True|False]` and then have `keys_length = 4 if include_none else 2`

Reply:
Sure thing. Parametrized this test over `include_none` instead